engine/examples/dotnetcore.md

```diff
@@ -0,0 +1,74 @@
+---
+description: Create a Docker image by layering your ASP.NET Core app on debian for Linux Containers or with Windows Nano Server containers using a Dockerfile.
+keywords: docker, dockerize, dockerizing, dotnet, .NET, Core, article, example, platform, installation, containers, images, image, dockerfile, build, asp.net, asp.net core
+title: Dockerizing a .NET Application
+---
+
+## Introduction
+
+This example demonstrates how to dockerize an ASP.NET Core application.
+
+> **Note:** This example assumes you already have a published ASP.NET Core app on your machine. If you are new to ASP.NET you can follow a [simple tutorial](https://www.asp.net/get-started) to initialize a project.
+
+## Why build ASP.NET Core?
+- [Open-source](https://github.com/aspnet/home)
+- Develop and run your ASP.NET Core apps cross-platform on Windows, Mac and Linux
+- Great for modern cloud-based apps, such as web apps, IoT apps and mobile backends 
+- ASP.NET Core apps can run on [.NET Core](https://www.microsoft.com/net/core/platform) or on the full [.NET Framework](https://www.microsoft.com/net/framework)
+- Architected to provide an optimized development framework for apps that are deployed to the cloud or run on-premises
+- Modular components with minimal overhead retain flexibility while constructing your solutions
+
+## Create a Dockerfile for an ASP.NET Core Application
+1. If you haven't already, publish your project locally by running `dotnet restore` and `dotnet publish -o published`.
+2. Create a `Dockerfile` in your project folder next to your published folder.
+3. Add the text below to your `Dockerfile` for either Linux or [Windows Containers](https://docs.microsoft.com/en-us/virtualization/windowscontainers/about/).
+4. The `Dockerfile` assumes that your application is called `aspnetapp`. Change the `Dockerfile` to use the .dll file of your project.
+
+  For Linux Containers:
+
+  ```dockerfile
+  FROM microsoft/aspnetcore:1.1
+  WORKDIR /app
+  COPY published ./
+  ENTRYPOINT ["dotnet", "aspnetapp.dll"]
+  ```
+
+  For Windows Containers:
+
+  ```dockerfile
+  FROM microsoft/dotnet:1.1-runtime-nanoserver
+  WORKDIR /app
+  ENV ASPNETCORE_URLS http://+:80
+  COPY published ./
+  ENTRYPOINT ["dotnet", "aspnetapp.dll"]
+  ```
+
+5. To make your build context as small as possible add a [`.dockerignore` file](https://docs.docker.com/engine/reference/builder/#dockerignore-file) to your project folder and copy the following into it.
```

Comment:
As in add the .git to be ignored or included? Isn't the `*` causing the context to ignore everything except the published folder that is excepted with the `!`?